File: C++ Programming/problems on file handling/problem9.cpp
/*
9. Write a program which accepts file name and one character from user and count number of occurrences of that characters from that file. 
	
	Input : Demo.txt  ‘M’ 
	
	Output : Frequency of M is 7 

	Author : Milind Nalawade

	Date : 1/09/2021

	Language : C++ programming
	
*/

using namespace std;
#include<iostream>
#include<stdlib.h>
#include<unistd.h>
#include<fcntl.h>

//Entry point function
int main()
{
	int fd=0,ret=0,cnt=0,i=0;
	char arr[100]={'\0'};
	char ch='\0';
	cout<<"\n Enter the file name : ";
	cin.getline(arr,100);
	cout<<"\n Enter your letter : ";
	cin>>ch;
	fd=open(arr,O_RDONLY|O_CREAT,0777);
	if(fd==-1)
	{
		cout<<"\n file unalbe to open\n";
	close(fd);
	}

	while((ret=read(fd,arr,100))!=0)
	{
		while(arr[i]!='\0')
		{
			if(arr[i]==ch)
			{
				cnt++;
			}
			i++;
		}			
	}
	cout<<"\n Frequency of "<<ch<<" is : "<<cnt;
	return 0;

}

File: Python Programming/problems on file handling/problem9.py
'''
9. Write a program which accepts file name and one character from user and count number of occurrences of that characters from that file. 
	
	Input : Demo.txt  ‘M’ 
	
	Output : Frequency of M is 7 

	Author : Milind Nalawade

	Date : 01/09/2021

	Language : Python 3.9.0
	
'''
import os
#=============
#Entry point function
#=============
def main():
	str=input("\n Enter file name : ")
	fd=open(str,"r")
	ch=input("\n Enter your character : ")
	if fd:
		print("-"*50)
		print(" File opened successfully !!!")
		print("-"*50)
	else:
		print(" File not found !!!")
	line=fd.read()
	i=cnt=0
	while i != len(line):
		if line[i]==ch:
			cnt+=1
		i+=1	
	print("\n Frequency of {} is : {} \n ".format(ch,cnt))	

#=====
#starter
#=====
if __name__=="__main__":
	main()
	